File: trunk/Core/libjausC/src/jausState.c
// File Name: jausState.c
//
// Written By: Danny Kent (jaus AT dannykent DOT com), Tom Galluzzo (galluzzo AT gmail DOT com)
//
// Version: 3.2
//
// Date: 08/04/06
//
// Description: 

#include <string.h>
#include <stdlib.h>
#include "cimar/jaus.h"

int jausStateToString(JausState state, char * buf)
{
	return sprintf(buf, "%s", jausStateGetString(state));
}

char *jausStateGetString(JausState state)
{
	static char *jausStateString[7] =	{	"Initialize",
											"Ready",
											"Standby",
											"Shutdown",
											"Failure",
											"Emergency",
											"Undefined" 
										};
	
	// If the state is unknown or out of bounds													
	if(state < JAUS_INITIALIZE_STATE || state > JAUS_EMERGENCY_STATE)
	{
		return jausStateString[6];
	}
	else
	{
		return jausStateString[state];
	}
}

File: openjaus/trunk/Core/libjausJava/src/jaus/JausState.java
package jaus;

public class JausState
{
	public static final int UNDEFINED	=	-1;
	public static final int UNKNOWN 	=	-1;
	public static final int INVALID 	= 	-1;

	public static final int INITIALIZE 	=	0;
	public static final int READY 		=	1;
	public static final int STANDBY 	=	2;
	public static final int SHUTDOWN 	=	3;
	public static final int FAILURE 	=	4;
	public static final int EMERGENCY 	=	5;
	
	int value;
	
	public JausState()
	{
		value = UNKNOWN;
	}

	public int getValue()
	{
		return value;
	}
	
	public void setValue(int value)
	{
		if(value > UNDEFINED && value <= EMERGENCY)
			this.value = value;
	}
	
	public String toString()
	{
		switch(value)
		{
			case INITIALIZE:
				return "Initialize";
				
			case READY:
				return "Ready";

			case STANDBY:
				return "Standby";

			case SHUTDOWN:
				return "Shutdown";

			case FAILURE:
				return "Failure";
			
			case EMERGENCY:
				return "Emergency";

			case UNDEFINED:
				return "Undefined/Unknown/Invalid";
				
			default:
				return "Undefined/Unknown/Invalid";
		}
	}
	
	public int hashCode()
	{
		return value;
	}
	
	public boolean equals(Object comparisionValue)
	{
		return value == ((JausState)comparisionValue).getValue();
	}
}
